src/containers/WebComponentLoader.test.js

```diff
@@ -25,106 +26,341 @@ const steps = [{ quiz: false, title: "Step 1", content: "Do something" }];
+      expect(cookies.cookies.theme).toEqual("light");
     });
   });
 
-  test("Enables the SenseHat", () => {
-    expect(store.getActions()).toEqual(
-      expect.arrayContaining([setSenseHatAlwaysEnabled(true)]),
-    );
+  describe("When props are set - logged in", () => {
+    beforeEach(() => {
+      localStorage.setItem(authKey, JSON.stringify(user));
+      render(
+        <Provider store={store}>
+          <CookiesProvider cookies={cookies}>
+            <WebComponentLoader
+              code={code}
+              identifier={identifier}
+              senseHatAlwaysEnabled={true}
+              instructions={instructions}
+              authKey={authKey}
+              theme="light"
+            />
+          </CookiesProvider>
+        </Provider>,
+      );
+    });
+
+    test("Calls useProject hook with correct attributes", () => {
+      expect(useProject).toHaveBeenCalledWith({
+        projectIdentifier: identifier,
+        code,
+        accessToken: "my_token",
+        loadRemix: false,
+        loadCache: true,
+        remixLoadFailed: false,
+      });
+    });
+
+    test("Calls useProjectPersistence hook with correct attributes", () => {
+      expect(useProjectPersistence).toHaveBeenCalledWith({
+        user,
+        project: { components: [] },
+        hasShownSavePrompt: false,
+        justLoaded: false,
+        saveTriggered: false,
+      });
+    });
+
+    test("Sets the instructions", () => {
+      expect(store.getActions()).toEqual(
+        expect.arrayContaining([setInstructions(instructions)]),
+      );
+    });
   });
 
-  test("Sets the instructions", () => {
-    expect(store.getActions()).toEqual(
-      expect.arrayContaining([setInstructions(instructions)]),
-    );
+  describe("When theme is not set", () => {
+    beforeEach(() => {
+      render(
+        <Provider store={store}>
+          <CookiesProvider cookies={cookies}>
+            <WebComponentLoader />
+          </CookiesProvider>
+        </Provider>,
+      );
+    });
+
+    test("Does not disable theming", () => {
+      expect(store.getActions()).not.toEqual(
+        expect.arrayContaining([disableTheming()]),
+      );
+    });
+
+    test("Does not set the theme", () => {
+      expect(cookies.cookies.theme).toBeUndefined();
+    });
   });
 
-  test("Disables theming", () => {
-    expect(store.getActions()).toEqual(
-      expect.arrayContaining([disableTheming()]),
-    );
+  describe("When local storage user changes", () => {
+    beforeEach(() => {
+      localStorage.setItem(authKey, JSON.stringify(updatedUser));
+      render(
+        <Provider store={store}>
+          <CookiesProvider cookies={cookies}>
+            <WebComponentLoader authKey={authKey} />
+          </CookiesProvider>
+        </Provider>,
+      );
+    });
+
+    test("Sets the user", () => {
+      expect(store.getActions()).toEqual(
+        expect.arrayContaining([setUser(updatedUser)]),
+      );
+    });
```

Comment:
Is this testing anything different from the tests where the user is set at the start? 🤔

Reply:
I've just moved this because I didn't wanted to test the user was updated but more that setUser was called and user was set in state from local storage